src/main/java/io/kestra/storage/azure/AzureStorage.java

```diff
@@ -89,81 +112,193 @@ public Long lastModifiedTime(String tenantId, URI uri) throws IOException {
         try {
             BlobContainerClient client = this.client();
 
-            ListBlobsOptions listBlobsOptions = new ListBlobsOptions();
-            listBlobsOptions.setPrefix(getURI(tenantId, storagePrefix).getPath());
-
-            List<String> deleted = Streams
-                .stream(client.listBlobs(listBlobsOptions, Duration.ofSeconds(30)))
-                .filter(blobItem -> blobItem.getProperties().getContentType() != null)
-                .map(throwFunction(itemResult -> {
-                    try {
-                        return itemResult.getName();
-                    } catch (Throwable e) {
-                        throw new IOException(e);
-                    }
-                }))
-                .collect(Collectors.toList());
-
-            deleted
-                .forEach(s -> {
-                    BlobClient blobClient = this.blob(URI.create(s));
-                    blobClient.delete();
-                });
-
-
-            return deleted
-                .stream()
+            String path = getPath(tenantId, storagePrefix);
+            if (!path.endsWith("/")) {
+                path += "/";
+            }
+            ListBlobsOptions listBlobsOptions = new ListBlobsOptions()
+                .setPrefix(path)
+                .setDetails(new BlobListDetails().setRetrieveDeletedBlobs(false).setRetrieveSnapshots(false));
+
+            List<String> deleted = new ArrayList<>();
+            List<String> directories = new ArrayList<>();
+            for (BlobItem itemResult : client.listBlobs(listBlobsOptions, Duration.ofSeconds(30))) {
+                String name = itemResult.getName();
+                Boolean isDir = this.client().getBlobClient(name + "/" + DIRECTORY_SUFFIX).exists();
+                if (isDir) {
+                    directories.add(name);
+                    continue;
+                }
+                BlobClient blobClient = this.client().getBlobClient(name);
+                blobClient.delete();
+                if (!name.endsWith(DIRECTORY_SUFFIX)) {
+                    deleted.add(name);
+                }
+            }
+
+            // also remove main directory (we have to remove start and ending /)
+            directories.add(path.substring(1, path.length() - 1));
+            directories.sort((s1, s2) -> s2.length() - s1.length());
+            for (String directory : directories) {
+                BlobClient blobClient = this.client().getBlobClient(directory);
+                blobClient.delete();
+                deleted.add(directory + "/");
+            }
+
+            return deleted.stream()
                 .map(s -> URI.create("kestra:///" + s.replace(tenantId + "/", "")))
-                .collect(Collectors.toList());
+                .toList();
         } catch (BlobStorageException e) {
+            if (e.getErrorCode() == BlobErrorCode.BLOB_NOT_FOUND || e.getErrorCode() == BlobErrorCode.RESOURCE_NOT_FOUND) {
+                return List.of();
+            }
```

Comment:
this seems to be the behavior of other storages. But as this is not standardized it might change as with the testsuit I'm aligning every storage to the localstorage one.